crates/bin/c8y_configuration_poc/src/config_manager/tests.rs

```diff
@@ -0,0 +1,292 @@
+use crate::c8y_http_proxy::messages::DownloadFile;
+use crate::c8y_http_proxy::messages::UploadConfigFile;
+use crate::config_manager::actor::OperationTimeout;
+use crate::config_manager::actor::OperationTimer;
+use crate::config_manager::ConfigManagerBuilder;
+use crate::config_manager::ConfigManagerConfig;
+use c8y_api::smartrest::topic::C8yTopic;
+use mqtt_channel::Topic;
+use std::time::Duration;
+use tedge_actors::Actor;
+use tedge_actors::Builder;
+use tedge_actors::SimpleMessageBoxBuilder;
+use tedge_mqtt_ext::MqttMessage;
+use tedge_test_utils::fs::TempTedgeDir;
+use tedge_utils::file::PermissionEntry;
+use tokio::time::timeout;
+
+const TEST_TIMEOUT: Duration = Duration::from_secs(5);
+
+#[tokio::test]
+async fn test_config_plugin_init() -> anyhow::Result<()> {
+    let device_id = "tedge-device".to_string();
+    let tedge_host = "127.0.0.1";
+    let c8y_host = "test.c8y.io";
+    let mqtt_port = 1234;
+    let tedge_http_port = 9876;
+
+    let test_config_type = "test-config";
+    let test_config_path = "/some/test/config";
+
+    let ttd = TempTedgeDir::new();
+    ttd.dir("c8y")
+        .file("c8y-configuration-plugin.toml")
+        .with_toml_content(toml::toml! {
+            files = [
+                { path = test_config_path, type = test_config_type }
+            ]
+        });
+
+    let config = ConfigManagerConfig::new(
+        ttd.to_path_buf(),
+        ttd.to_path_buf(),
+        device_id.clone(),
+        tedge_host.to_string().try_into().unwrap(),
+        mqtt_port,
+        c8y_host.to_string().try_into().unwrap(),
+        tedge_host.to_string().try_into().unwrap(),
+        tedge_http_port,
+    );
+
+    let mut mqtt_builder: SimpleMessageBoxBuilder<MqttMessage, MqttMessage> =
+        SimpleMessageBoxBuilder::new("MQTT", 5);
+    let mut c8y_proxy_builder: SimpleMessageBoxBuilder<C8YRestRequest, C8YRestResult> =
+        SimpleMessageBoxBuilder::new("C8Y", 1);
+    let mut timer_builder: SimpleMessageBoxBuilder<OperationTimer, OperationTimeout> =
+        SimpleMessageBoxBuilder::new("Timer", 5);
+
+    let mut config_manager_builder = ConfigManagerBuilder::new(config);
+
+    config_manager_builder.with_c8y_http_proxy(&mut c8y_proxy_builder)?;
+    config_manager_builder.with_mqtt_connection(&mut mqtt_builder)?;
+    config_manager_builder.with_timer(&mut timer_builder)?;
+
+    let mut mqtt_message_box = mqtt_builder.build();
+
+    let (actor, message_box) = config_manager_builder.build();
+    let _join_handle = tokio::spawn(async move { actor.run(message_box).await });
+
+    // Assert supported config types MQTT message
+    let expected_message = MqttMessage::new(
+        &C8yTopic::SmartRestResponse.to_topic().unwrap(),
+        format!("119,c8y-configuration-plugin,{test_config_type}"),
+    );
+    let next_message = timeout(TEST_TIMEOUT, mqtt_message_box.recv())
+        .await?
+        .unwrap();
+    assert_eq!(next_message, expected_message);
```

Comment:
Rather than an unwrap that might hide errors, I would explicitly compare the message to some expected one.

```suggestion
    let next_message = timeout(TEST_TIMEOUT, mqtt_message_box.recv()).await;
    assert_eq!(next_message, Ok(Some(expected_message)));
```